### PR TITLE
Progress table container unit tests

### PR DESCRIPTION
**Pull request by maureensturgeon**
Add progress table container unit tests

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx [hunk 1]
```diff
@@ -46,7 +46,9 @@ class ProgressTableContainer extends React.Component {
     lessonOfInterest: PropTypes.number.isRequired,
     levelProgressByStudent: PropTypes.objectOf(
       PropTypes.objectOf(studentLevelProgressType)
-    ).isRequired
+    ).isRequired,
+    studentTimestamps: PropTypes.object,
```
Comment (maureensturgeon) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx:50]: Should this be required?

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableContainerTest.jsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  });
+
+  it('passes needsGutter true to the ProgressTableContentView when the student row height exceeds the body height', () => {
+    // 18 students will exceed max height
+    const students = Array(18)
+      .fill()
+      .map((x, i) => ({id: i, name: `student-${i}`}));
+
+    const wrapper = setUp({section: {id: 1, students: students}});
+    expect(wrapper.find(ProgressTableContentView).props().needsGutter).to.equal(
```
Comment (jamescodeorg) [inline apps/test/unit/templates/sectionProgress/progressTables/ProgressTableContainerTest.jsx]: I think `to.be.true` is the more canonical way to check this condition?  Similar comments for other boolean checks in this file.  (I'm not sure though, I'm just learning chai too.)

File: apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx [hunk 3]
```diff
@@ -46,7 +46,9 @@ class ProgressTableContainer extends React.Component {
     lessonOfInterest: PropTypes.number.isRequired,
     levelProgressByStudent: PropTypes.objectOf(
       PropTypes.objectOf(studentLevelProgressType)
-    ).isRequired
+    ).isRequired,
+    studentTimestamps: PropTypes.object.isRequired,
```
Comment (jamescodeorg) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx:50]: Are these written to by other parts of Charlie's changes?  (It's ok if they are, just wanted to double-check.)

**Review by jamescodeorg — APPROVED** [review]

File: apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx (hunk 3 above)
Comment (maureensturgeon) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx:50]: These are props coming in through redux that were missing from propTypes, which I discovered while writing tests. By written to, did you mean passed into the component?

Comment (cforkish) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx:50]: @jamescodeorg this component subscribes to redux props needed by its children to avoid lots of duplicated work in `connect`. originally i was only validating prop types in the components that actually use the props, but we decided that it makes sense to validate pass-through props even though they're only needed in the children. does that make sense? and do you think this pattern (of validating pass-through props) makes sense?

File: apps/test/unit/templates/sectionProgress/progressTables/ProgressTableContainerTest.jsx (hunk 2 above)
Comment (maureensturgeon) [reply apps/test/unit/templates/sectionProgress/progressTables/ProgressTableContainerTest.jsx]: Yup, you're right, I'll update those statements!

File: apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx (hunk 3 above)
Comment (jamescodeorg) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx:50]: Oh, interesting.  I agree with this in principle, but in practice, does this mean that a component near the top of tree might end up validating a lot of properties?  Is there an accepted best practice around validating prop types in redux?  (To be honest, I still don't totally grok the principles behind redux.  Naively, it feels kind of like a giant global variable store with some data flow constraints and good debugging tools.)

Comment (cforkish) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx:50]: a quick search for best practice about this only produced this one not-very-definitive answer: https://stackoverflow.com/questions/49594216/should-i-use-propttypes-for-props-i-pass-through-a-react-component

tl;dr, one person says validate them, one person says don't, and one person offers another way to validate them by requiring `ChildComponent.propTypes`. since many of the pass-through props are used by multiple children, i don't think that last option makes sense here. so my vote is to leave this as-is.

@jamescodeorg your feeling about redux in our codebase is spot on. however, i believe that's just because we're not using it as intended. i met with Dave during my second week on the job to discuss this. unfortunately fixing this issue is going to require some major rearchitecting, but it's still my hope to start moving in that direction as part of our "north star" discussions during H1.

**Review by cforkish — APPROVED** [review]

File: apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx (hunk 3 above)
Comment (jamescodeorg) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx:50]: Ok, thanks for doing the research.  I don't have a strong opinion (yet? :-) so whatever you think is best sounds good to me.
